flocker/node/agents/blockdevice.py

```diff
@@ -230,6 +235,46 @@ def _action(self):
         return DESTROY_BLOCK_DEVICE_DATASET(_logger, volume=self.volume)
 
 
+# Implement:
+#     AttachVolume(volume=volume),
+#     MountVolume(volume=volume),
+# in order to carry out the sequence below.
+# Maybe use those in the CreateBlockDeviceDataset state change or paste in some
+# followup issues, or maybe do those issues first??
+
+
+@_logged_statechange
+@with_cmp(["change"])
+class ResizeBlockDeviceDataset(proxyForInterface(IStateChange, "change")):
+    """
+    Resize the volume for a dataset with a primary manifestation on the node
+    where this state change runs.
+
+    :ivar IStateChange change: The real implementation of this state change.
+    :ivar volume: See ``__init__``.
+    """
+    def __init__(self, volume, size):
+        """
+        :param BlockDeviceVolume volume: The volume which will be resized.
+        :param int size: The required size of the volume
+        """
+        # self.volume = volume
+        # sequence = Sequentially(changes=[
+        #     UnmountBlockDevice(volume=volume),
+        #     DetachVolume(volume=volume),
+        #     ResizeVolume(volume=volume, size=size),
+        #     AttachVolume(volume=volume),
+        #     MountVolume(volume=volume),
+        # ])
+        # super(ResizeBlockDeviceDataset, self).__init__(sequence)
+        pass
+
+
+    @property
+    def _action(self):
```

Comment:
This is a *super* confusing name for the method. "_eliot_action" maybe?

Or better yet get rid of `_logged_statechange` and just do https://clusterhq.atlassian.net/browse/FLOC-1591 before this. Yes, that won't give the best logging ever, but it'll give a decent baseline and the current approach seems like it's setting you up for lots of boilerplate copy/paste testing.